storage/rocksdb/rdb_datadic.cc

```diff
@@ -3137,6 +3147,20 @@ bool Rdb_tbl_def::put_dict(Rdb_dict_manager *const dict,
   return false;
 }
 
+uint32 Rdb_key_def::calculate_index_flag_offset(uint32 index_flags,
+                                                enum INDEX_FLAG flag) {
+  uint offset = 0;
+  uint flag_len = 0;
+  if (index_flags & Rdb_key_def::TTL_FLAG) {
+    offset += ROCKSDB_SIZEOF_TTL_RECORD;
```

Comment:
Spoke about this offline. Could change this to use an array that has all of the sizes of each of the flag types. Then iterate through index_flags to determine if a particular flag is set. If it is, then add the size of the value to the offset. Return offset when the iteration level == flag. Could assert flag & index_flags somewhere for correctness, but its possible future flag types might not exist in index_flags, so this needs to handle that condition. Returning UINT_MAX for the not found offset is probably fine, but all callers to this function would need to check/assert on it.